src/main.rs

```diff
@@ -164,7 +164,14 @@ async fn main() -> Result<(), Box<dyn Error>> {
     );
     tokio::spawn(async move { mempool.run().await });
 
-    let admin_service = MyAdminService::new(db_manager, mempool_tx.clone());
+    let admin_service = MyAdminService::new(
```

Comment:
note, we're calling backup_db here in main. I think it's worth keeping that but maybe using an explicit config variable like backup_on_startup or similar.

Reply:
What's the point of backing up on startup? Right now, there's very likely going to be nothing new in the db right?

Reply:
I added the flag and left this code as-is. Left a comment that we might want to reconsider this, especially when we schedule snapshots.